### PR TITLE
Ensure numerical aquifer cells get the correct depth

### PR DESCRIPTION
**Pull request by vkip**
Should resolve https://github.com/OPM/opm-simulators/issues/4222

### CODE REVIEW

**Comment by vkip** [comment]
jenkins build this please

File: src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp [hunk 1]
```diff
@@ -1678,6 +1678,12 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
 
     double EclipseGrid::getCellDepth(size_t globalIndex) const {
         assertGlobalIndex( globalIndex );
+
+        auto it = this->m_aquifer_cell_depths.find(globalIndex);
+        return it != this->m_aquifer_cell_depths.end() ? it->second : computeCellDepth(globalIndex);
+    }
+
+    double EclipseGrid::computeCellDepth(size_t globalIndex) const {
```
Comment (atgeirr) [inline src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1686]: I suggest renaming this `computeCellGeometricDepth()` or similar, to make clearer the difference between this and `getCellDepth()`.

**Comment by vkip** [comment]
jenkins build this please

**Comment by vkip** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp [hunk 2]
```diff
@@ -30,7 +30,7 @@
 #include <optional>
 #include <stdexcept>
 #include <unordered_set>
-#include <vector>
```
Comment (atgeirr) [inline opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp:33]: This line should also be kept I think...

Comment (vkip) [reply opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp:33]: Doh.. Thanks!

**Comment by vkip** [comment]
jenkins build this please

**Comment by vkip** [comment]
Sorry about the deleted lines mess.. Now we're back to the one failing test (3D_1AQU_3CELLS), where I believe the new behavior (using aquifer depth) is the correct one.

**Comment by atgeirr** [comment]
> Sorry about the deleted lines mess.. Now we're back to the one failing test (3D_1AQU_3CELLS), where I believe the new behavior (using aquifer depth) is the correct one.

I believe you... I will start an update_data run. Before merging, I would appreciate if you could squash the commits.

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Review by bska — APPROVED** [review]
I agree that this is necessary.  It might not, however, be fully sufficient.  For what it's worth, I believe the equilibration facility infers cell centre point depths from the cell geometry, and that will possibly lead to an inconsistent initial pressure value in numerical aquifer cells when the initial pressure item in `AQUNUM` is defaulted (common case).

I think we'll have to review that code quite carfully to ensure that it does what it's supposed to do.  For now, this is a clear improvement, so I'll update the reference solutions and merge this into the master branch.

**Comment by bska** [comment]
jenkins build this opm-tests=1046 please

**Comment by bska** [comment]
The new reference solutions have been installed on the CI system.  I'll merge this now to correct the underlying issue.

**Pull request merged**